### ISSUE TITLE
building computeKey cache on first `fcu` call

### ISSUE THREAD

**Issue by advaita-saha** [status-im/nimbus-eth1#2748]
https://github.com/status-im/nimbus-eth1/blob/0b93236d1b6d7436e92ed39d44ac25008326a0f6/nimbus/config.nim#L531-L535

The default mode should be with validation, will also smooth out the UX from `import` to `syncer`. As with validation the triggering of the computeKey cache won't take place with the first `fcu` call which takes hours and results in the `fcu` call being timed out

Suggestion: `defaultValue: false`

**Comment by arnetheduck** [comment]
Changing the default here increases the total time it takes to import + compute cache - the way forward here is _probably_ to check that a cache is available on normal (non-import) startup or something similar to that effect, so that it is computed before turning on networking.

**Comment by advaita-saha** [comment]
Yes there is slow down, but personally as far as I ran imports for holesky and sepolia. The slowdown is not very significant.  But yes the approach for checking on start does makes sense.

**Comment by arnetheduck** [comment]
> The slowdown is not very significant. 

you can measure it to quantify this point, but that only makes sense to do so at mainnet scale, ie around block 20m where the database no longer fits in memory.

**Comment by arnetheduck** [comment]
https://github.com/status-im/nimbus-eth1/pull/2788 addresses this somewhat

>  The slowdown is not very significant. 

a significant part of the slowdown is due to the separation of vertex and key data in the database - when the database is big, this leads to a lot of read/write amplification as each mpt node has to be read from two tables - it's worth remembering for future reference however - if we were to join the two tables, it might very well be the case that computing during import is decently fast.

**Comment by advaita-saha** [comment]
Not relevant anymore
